First pass of Applied Energistics 2 emissives retexturing

QuantumManiac opened the pull request:
I did a bit of work on adding emissives to various AE2 blocks and items wherever I saw fit and was possible. While I did retexture a lot of blocks, there are still some remaining that were not retextured either because I didn't know if it needed emissives or due to technical difficulties due to how the textures were made. More on this in their respective sections below.

Here's what this branch has retextured so far, along with some comments about it.

Shaders, if used in screenshot, is BSL v7.1.04.1, with modified lighting to make light sources a less warm color. MC Version is 1.12.2.

_Yes, I did just copy/paste this pre-amble from the previous pull request._

**ME Chest**
![image](https://user-images.githubusercontent.com/8643561/82739067-a879a680-9d0a-11ea-856a-cab31e60e8c1.png)
Emissive on storage indicator and screen when a cell is inside and powered.

**Crafting Unit**
![image](https://user-images.githubusercontent.com/8643561/82738806-85e68e00-9d08-11ea-8801-0afff74dec44.png)
Not that emissives exist even when the block is not powered. However, there are no emissives on the crafting monitor until the multiblock is valid.

**Spatial Pylon**
![image](https://user-images.githubusercontent.com/8643561/82739055-93047c80-9d0a-11ea-894b-df6da837101d.png)
I was too lazy to get a proper frame going to showcase the valid spatial pylons so you get single blocks instead.

**Charged Certus Quartz**
![image](https://user-images.githubusercontent.com/8643561/82739046-83853380-9d0a-11ea-8c58-385c55ac95c5.png)
Quite sure this emissive breaks with Optifine too

**Charger**
![image](https://user-images.githubusercontent.com/8643561/82739038-78320800-9d0a-11ea-9854-6dbf025e91e3.png)
Just a bit of subtle lighting

**Matter Condenser**
![image](https://user-images.githubusercontent.com/8643561/82739027-6a7c8280-9d0a-11ea-89fb-a238e0359aa5.png)


**ME Controller**
![image](https://user-images.githubusercontent.com/8643561/82739019-5f295700-9d0a-11ea-8186-c21bb3ce7b76.png)

Done originally. 

**Energy Cells**
![image](https://user-images.githubusercontent.com/8643561/82739013-53d62b80-9d0a-11ea-8b4e-6f4297582489.png)

No emissives when at zero charge

**ME Drive and Energy Acceptor**
![image](https://user-images.githubusercontent.com/8643561/82739010-4a4cc380-9d0a-11ea-828a-629b94ed4678.png)


**ME and Spatial IO Ports**
![image](https://user-images.githubusercontent.com/8643561/82739005-330dd600-9d0a-11ea-8b7d-2434b41d85b9.png)

**Molecular Assembler**
![image](https://user-images.githubusercontent.com/8643561/82739110-f393b980-9d0a-11ea-8a7e-a4bb0bd3f5ff.png)
Rainbow light cycling works perfectly!

**Quantum Link Chamber/Ring**
![image](https://user-images.githubusercontent.com/8643561/82739142-2ccc2980-9d0b-11ea-9b3f-ecfe7b3909d1.png)
Emissives only work when multiblock is constructed and powered.

**ME Security Terminal**
![image](https://user-images.githubusercontent.com/8643561/82739200-aebc5280-9d0b-11ea-9dfa-84e9efd96f88.png)
Terminal screen is lit regardless of whether it is powered or connected. Need to look into a fix

**Wireless Access Point**
![image](https://user-images.githubusercontent.com/8643561/82739245-170b3400-9d0c-11ea-990b-7ac2ed7f3cfd.png)
Only lit up when powered

**Skystone Stuff**
![image](https://user-images.githubusercontent.com/8643561/82739264-3d30d400-9d0c-11ea-9e99-66bb55af781f.png)
Added same emissives as blocks for the chests. Doesn't look too good in my opinion, might be worth doing a second pass of?

**Vibration Chamber**
![image](https://user-images.githubusercontent.com/8643561/82739297-84b76000-9d0c-11ea-809d-a4ab3d1f70bd.png)
Lights up when burning

**Items**
![image](https://user-images.githubusercontent.com/8643561/82739405-48383400-9d0d-11ea-9ad8-ff24062675a6.png)
More than these had emissives added to them, I just didn't want a dozen item frames JUST for storage cells. Due to limitations, items light up even if they're at zero energy (e.g. Wireless Terminal)

**Smart Cable + P2P Tunnels**
![image](https://user-images.githubusercontent.com/8643561/82739479-d4e2f200-9d0d-11ea-9dfb-40e633fb2fc6.png)
All the channel indicators for the smart cables light up, as well as the frequency indicator for the P2P tunnels.

There is a known bug(?) where the channel indicator for the smart cables disappear once you're ~30 blocks away from the cable.

And that's about it for what I thought needed emissives.


**Additonal thoughts/notes:**

- Currently, it seems that the ME Terminals don't really "light up" when they are connected to an ME system. Because of this, I decided not to add emissives to the terminals as they would be super bright even when not connected. Feedback appreciated.

- It would be cool to make the blue parts of the Cell Workbench light up, although I'm not sure what that material is supposed to be.

- Adding emissive textures for the particles (e.g. Matter Cannon shots) has no effect

- in the \items\ directory, there's a texture named "white.png". I'd like to figure out what that's for.



Finally, I suggest that we Don't Merge this pull request unless all the issues that exist are addressed. Alternatively, we can merge just the blocks that don't have any issues not and merge the rest later, once they are fixed.

QuantumManiac committed:
Add initial AE2 work

QuantumManiac committed:
Retextured various items, most notably channel indicators on smart cables

Liamdude commented:
I read this at the same time as the other post, but just didn't respond at the time. These look very good overall. It is a shame that some things will still light up even when not powered, but there really isn't a great solution for that other than just not having them be emissive at all.

I would agree on the ME Terminals not being emissive because it would be odd for things to show light without having power that would supply that light.
The cell workbench could be fine with emissive textures. The idea is likely that the blue is a visible computerized component. Otherwise, the recipe is not too helpful because one could argue that the visible components are actually wool. I would test it and see how it looks in practice, but it doesn't have the same color brightness that other lighted items do in this mod.

I am not sure what the white png is for, likely an overlay of some sort. I'll have to check the files.

Items look very good. Skystone dust looks very good here in particular. However, I am not sure that the paintball (I believe that is in the bottom row, second from the right) should be emissive at all. I would also argue that the wireless terminal should only have the antenna lit. Having the whole device lit seems unnecessary, and again, its lit color does not match many of the others in this mod. Another option might be to play with having the purple buttons and the white/gray lit/unlit both ways to see if those present a better visual. For the singularities, I would not make the borders emissive: the color is too dark.

It's funny you mention the skystone. When I first started this resource pack, I played without shaders, and I thought the skystone looked fine. I now use BSL and in a new world I started, I spawned right by skystone, and the current design did look rather unsightly. I assume that has something to do with the way lighting works in a shader, or I might be sugarcoating how I thought this texture looked in vanilla. The overall idea was that foreign matter would probably have a glow, and it's space material, which makes sense to have a glow or sheen. That said, I might revisit those textures because they do look a bit strange. Maybe just reducing the number of pixels that are emissive. Maybe.

I am not sure how feasible it is to try fixing the textures when they are lit despite not being powered. Depends on whether the texture files are separate based on powered versus not powered. If there is no difference, then they can likely be left as not emissive because they would be odd otherwise.

For the charged quartz, I would also remove the outer edge. Having that be emissive is strange, especially because that portion is where the quartz fades back into the rock face.

Based on what I am seeing, the spatial pylons look too emissive. The borders should not be nor should the darker glow sections in the corners of the interior (wow, that is a terrible description). I would, at most, light only the center, first ring, and second ring of the pylons. Even then, maybe only the center and the first ring because the texture is naturally meant to show some sort of radiating color. Making it all emissive defeats the purpose because now even the lesser glow gives off substantial glow. I would also test these to see how they look with less emissive glow.

QuantumManiac commented:
Yeah, I thought for a while about whether or not the ME terminals should or should not have always-on emissive textures. I argued that the terminals would usually be always on anyway, or that they have internal power or something of the sorts but that wasn't convincing enough. In the end, I realized that it was a choice between realism/immersiveness and aesthetic, and I went for the former. 

The paintballs shown are the "Lumen" variant of paintballs. They're normal paintballs combined with glowstone which lead me to believe that you can use them to make glowing paint or something of the sorts. The regular paintballs do not have emissives. As for the wireless terminal, I added emissives to only the antenna and the terminal display because that was what made sense, really. I will look into making it look a bit less bright though. I do agree with you on the singularities though, and will change them ASAP.

I compared the look of the skystone block with both BSL and no shaders and I agree that it does look quite a bit different with shaders on. My suggestion to improve it likely would be to keep the same emissive particles on the texture, but darken them a bit to make the glow more subtle. I'll whip up a concept if I get a chance

I do agree with the charged quartz too, now that I'm looking at it a bit more. I found a texture layer that seemed to be the layer that produced the Forge lighting emissives with the original textures so I just appended an "_e" and didn't think much more about it.

Yeah, I understand what you mean about the spatial pylons. I only added emissives to the first and second rings like you have suggested but I think the issue stems from the fact that the pylons emit light. The decrease in contrast due to all the stone below the blocks being lit up does make it look like everything is emissive.




Thanks for the feedback once again though!

QuantumManiac committed:
Various changes to AE2 emissives

- Removed emissives for dark borders on singularity items
- Removed excess emissives on charged certus quartz ore blocks
- Reduced emissives on spatial pylons

Liamdude commented:
Going to merge these, since I think these are where they should be.
